src/Umbraco.Core/Persistence/Repositories/Implement/UserRepository.cs

```diff
@@ -557,6 +557,16 @@ protected override void PersistUpdatedItem(IUser entity)
                 }
             }
 
+            // If userlogin or the email has changed then need to reset security stamp
+            if (changedCols.Contains("userLogin") || changedCols.Contains("userEmail"))
```

Comment:
Looks good, all that's missing is a unit test :) Should be an easy one to write in the UserRepositoryTests, you can have a look at `Can_Perform_Update_On_UserRepository` for inspiration, but for this one we'll just be testing/asserting this logic.